src/pages/insights/aws-config-compliance-automation.mdx

```diff
@@ -0,0 +1,107 @@
+
+## Alternative solution
+
+In these cases, some customization is necessary to drive visibility of the non-compliant items from AWS Config into the normal resolution workflow in ServiceNow. A solution that is possible, is to pull data from AWS Config using `aws-sdk` API's, and then feed that information into ServiceNow using ServiceNow API's.
+
+Many organizations deploy resources into several AWS accounts, making management of these items a little trickier. Luckily, AWS Config provides a capability to track compliance data at an <a href="https://docs.aws.amazon.com/config/latest/developerguide/aggregate-data.html">aggregate-level, across many accounts</a>. Once aggregated, the data is made available in a few different ways. The most flexible of which seems to be utilizing <a href="https://docs.aws.amazon.com/config/latest/developerguide/querying-AWS-resources.html">AWS Config Advanced Queries</a> to pull back compliance information across many accounts while utilizing the same method to get the data for each. 
+
+A sample query used from the aggregator to get non-compliant information from a single account could look like the following:
+
+```sql
+SELECT
+  resourceId,
+  configuration.targetResourceType,
+  configuration.complianceType,
+  configuration.configRuleList
+WHERE
+  accountId = '<account-id>'
+  AND configuration.complianceType = 'NON_COMPLIANT'
+```
+
+which pulls back the offending resource, its type, and information about which AWS Config rules the resource was shown to be non-compliant for.
+
+Once this information is retrieved, it can be compared against previous known state to determine if the compliance finding is new (and needing ServiceNow tracking) or has been previously reported (and subsequently tracked in ServiceNow). It can also be used to determine if a resource, which was previously reported as non-compliant, is now compliant (say for instance, if someone in IT has manually intervened to reconfigure the resource such that it now complies with configuration rules). If the resource is back in compliance, the correlated ServiceNow item used to track the offense can be successfully closed.
+
+One way to help assist with all of these abilities, is to create an AWS Step Function surrounding the process. AWS Step Functions can document and help coordinate the workflows of tasks in many areas of AWS, but a very typical workflow can be orchestrated by pulling together AWS Lambda Functions into a coordinate set of tasks. 
+
+Storing metadata around the process is helpful too, and any persistence store desired can be used. AWS DynamoDB lends itself very well to simple data storage needs, and a sample set of information stored in DynamoDB which could assist a process such as compliance automation might look something like:
+
+| Field      | Description |
```

Comment:
Might be nice if there was more spacing after tables. 
![image](https://user-images.githubusercontent.com/8344081/234134791-727591c5-bdad-400e-b878-5866b8743839.png)

Reply:
To be fair, this blog post has been a bear. I blame @larryboymi